design/tls-certificate-delegation.md

```diff
@@ -0,0 +1,86 @@
+# TLS Certificate Delegation
+
+_Status_: Draft, in review.
+
+This document outlines a specification to allow an Ingress Controller to reference a TLS certificate stored in one namespace from an Ingress or IngressRoute document stored in another namespace.
+
+## Goals
+
+- Permit wildcard TLS certificates to be referenced by Ingress/IngressRoute objects in other namespaces.
+
+## Non-goals
+
+- To implement a notion of a default, or fallback, TLS certificate available to all Ingress/IngressRoute objects.
+- To permit Secrets to be referenced from any Kubernetes object.
+- To permit other classes of Kubernetes objects to be referenced across namespaces.
+
+## Background
+
+Currently the Secret containing the TLS certificate must be co-located in the same namespace as the Ingress or root IngressRoute object referencing that secret.
+This requirement complicates deployment patterns where wildcard TLS certificates are used, specifically the use of a wildcard certificate to secure a number of subdomains where the Ingress/IngressRoute records for those subdomains do not share the same namespace as the wildcard TLS certificate.
+For example, presenting foo.example.com using the certifiate for \*.example.com when the Ingress/IngressRoute object for foo.example.com does not share the same namespace as the Secret containing \*.example.com.
+
+This proposal introduces a specification, in the form of a simple CRD, whereby the permission to access the contents of a secret containing a TLS certificate is delegated from the owning namespace to one or more other namespaces.
+
+## High-Level Design
+
+- The definition of a Secret's name is extended to recognize a namespace prefix, ie. `secretName: kube-system/wildcard-tls` means the secret `wildcard-tls` in the `kube-system` namespace.
+- A `TLSCertificateDelegation` CRD grants the permission to reference the contents of a Secret in the owner's namespace to an Ingress controller operating in the context of an Ingress or IngressRoute object from a different namespace for the purpose of retrieving the TLS certicate.
+
+## Detailed Design
```

Comment:
@davecheney could you also provide additional details on how the `status` field in IngressRoute would be changed in the case of missing (orphaned?) delegation or any other failure conditions.